homeassistant/components/dwd_weather_warnings/util.py

```diff
@@ -0,0 +1,32 @@
+"""Util functions for the dwd_weather_warnings integration."""
+
+from __future__ import annotations
+
+from homeassistant.core import HomeAssistant
+
+from .const import LOGGER
+
+
+def get_position_data(
+    hass: HomeAssistant, device_tracker: str
+) -> tuple[float, float] | None:
+    """Extract longitude and latitude from a device tracker."""
+    entity = hass.states.get(device_tracker)
+    if entity is None:
+        return None
+
+    latitude = entity.attributes.get("latitude")
+    if not latitude:
+        LOGGER.warning(
```

Comment:
Don't log in the util. Let the caller handle that as needed.

Reply:
Updated in commit 1bec061985b3573c7b3089ea62d17edda8422a4a